shared/login/signup/passphrase/index.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import * as React from 'react'
+import * as Constants from '../../../constants/signup'
+import {Avatar, Input, WaitingButton} from '../../../common-adapters'
+import {isMobile} from '../../../styles'
+import Wrapper from '../wrapper'
+
+type Props = {|
+  passphrase: string,
+  error: string,
+  onBack: () => void,
+  onSubmit: (pass1: string, pass1: string) => void,
+|}
+type State = {|
+  pass1: string,
+  pass2: string,
+|}
+
+class Passphrase extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props)
+    this.state = {pass1: this.props.passphrase, pass2: this.props.passphrase}
+  }
+  _onSubmit = () => {
+    this.props.onSubmit(this.state.pass1, this.state.pass2)
+  }
+  render() {
+    return (
+      <Wrapper onBack={this.props.onBack}>
+        <Avatar username="" size={isMobile ? 96 : 128} />
+        <Input
+          autoFocus={true}
+          hintText="Create a passphrase"
+          value={this.state.pass1}
+          type="password"
+          errorText={this.props.passphrase === this.state.pass1 ? this.props.error : ''}
```

Comment:
This leads to a weird behavior, when changing the passphrase if you enter the original one the error flashes back in. Maybe `onChangeText` should clear the error if one is coming in through props.